plugins/module_utils/template.py

```diff
@@ -440,3 +441,12 @@ def get_route_map_policy_for_multicast_uuid(self, route_map_policy_for_multicast
         kv_list = [KVPair("name", route_map_policy_for_multicast_name)]
         match = self.get_object_by_key_value_pairs("Route Map Policy for Multicast", existing_route_map_policies, kv_list, fail_module=True)
         return match.details.get("uuid")
+
+    def get_template(self, template_type, template_name, template_id):
+        if template_id in self.cache:
+            return self.cache[template_id]
+
+        new_template = MSOTemplate(self.mso, template_type, template_name, template_id)
+        self.cache[new_template.template_id] = new_template
+        self.cache[(new_template.template_name, new_template.template_type)] = new_template
```

Comment:
Should the name/type tuple key be checked above if template_id isn't provided?

Reply:
Done!